### ISSUE TITLE
AzureQueueStreamProvider errors after trying to publish message

### ISSUE THREAD

**Issue by charleszipp** [dotnet/orleans#3056]
I am exploring orleans for a potential alternative to ServiceFabric actors. I am testing orleans 1.4.1 on service fabric. I have been researching the streams.
I was able to get the SMSProvider working but, the same publishing and subscription does not seem to work with AzureQueue. I am getting the following error. Also, the message is never received by the grain. No exception is thrown by the client producing the message.
```
ERROR 103317 PullingAgentSystemTarget/FF/921360bb-AZQProvider 127.0.0.1:30036] !!!!!!!!!! Ignored RegisterAsStreamProducer Error Exc level 0: System.ArgumentException: Trying to register non active IStreamProducerExtension: SystemTarget:PullingAgentSystemTarget/FF/921360bb/S127.0.0.1:30036:233499092 Parameter name: streamProducer
```

I feel like i may be missing something in the setup or, the persistent streams require a few more steps than the SMSProvider.

Stateless Service hosting the silo
```csharp
internal sealed class OrleansStatelessService : StatelessService
{
    public OrleansStatelessService(StatelessServiceContext context)
        : base(context)
    { }

    public ClusterConfiguration GetClusterConfiguration()
    {
        var config = new ClusterConfiguration();

        config.Globals.ReminderServiceType = GlobalConfiguration.ReminderServiceProviderType.ReminderTableGrain;
        config.Globals.DataConnectionString = "UseDevelopmentStorage=true";
        config.Defaults.StartupTypeName = typeof(ClusterStartup).AssemblyQualifiedName;
        config.AddAzureTableStorageProvider("Default");
        config.AddAzureTableStorageProvider("PubSubStore");
        config.AddAzureQueueStreamProvider("AZQProvider", deploymentId: Regex.Replace(this.Context.ServiceName.PathAndQuery.Trim('/'), "[^a-zA-Z0-9_]", "").ToLower());
            
        LogManager.LogConsumers.Add(new EventSourceLogger());
        return config;
    }

    protected override IEnumerable<ServiceInstanceListener> CreateServiceInstanceListeners()
    {
        ClusterStartup.Service = this;
        return new[] { OrleansServiceListener.CreateStateless(this.GetClusterConfiguration()) };
    }

    protected override async Task RunAsync(CancellationToken cancellationToken)
    {
        while (true)
        {
            cancellationToken.ThrowIfCancellationRequested();
            await Task.Delay(TimeSpan.FromSeconds(10), cancellationToken);
        }
    }
}
```

Console producing the messages
```csharp
internal class Program
{
    private static void Main(string[] args)
    {
        Console.WriteLine("Press any key once service is ready...");
        Console.ReadKey();

        var serviceName = new Uri("fabric:/Audit.OrleansTest/Audit.OrleansTest.Service");
        var config = new ClientConfiguration
        {
            DeploymentId = Regex.Replace(serviceName.PathAndQuery.Trim('/'), "[^a-zA-Z0-9_]", "_"),
            GatewayProvider = ClientConfiguration.GatewayProviderType.Custom,
            CustomGatewayProviderAssemblyName = typeof(OrleansServiceFabricExtensions).Assembly.FullName,
            DataConnectionString = "fabric:/Audit.OrleansTest/Audit.OrleansTest.Service",
        };
        config.AddAzureQueueStreamProvider("AZQProvider", "UseDevelopmentStorage=true", deploymentId: Regex.Replace(serviceName.PathAndQuery.Trim('/'), "[^a-zA-Z0-9_]", "").ToLower());
        GrainClient.Initialize(config);
        var id = Guid.NewGuid();
        var streamProvider = GrainClient.GetStreamProvider("AZQProvider");            
        var stream = streamProvider.GetStream<string>(id, "TESTDATA");
        stream.OnNextAsync("testing123").Wait();
        Console.WriteLine($"Published to stream TESTDATA id {id}");
        Console.ReadKey();
    }
}
```

Grain subscribing to messages
```csharp
[ImplicitStreamSubscription("TESTDATA")]
public class EngagmentGrain : Grain, IEngagementGrain, IAsyncObserver<string>
{
    protected StreamSubscriptionHandle<string> _sub;

    public Task<string> Test(string input)
    {
        return Task.FromResult($"you said {input}. i say hello!");
    }

    public override async Task OnActivateAsync()
    {
        var guid = this.GetPrimaryKey();
        var streamProvider = GetStreamProvider("AZQProvider");
        var stream = streamProvider.GetStream<string>(guid, "TESTDATA");
        _sub = await stream.SubscribeAsync(this);            
    }

    public override async Task OnDeactivateAsync()
    {
        if (_sub != null)
            await _sub.UnsubscribeAsync();
    }

    public Task OnNextAsync(string item, StreamSequenceToken token = null)
    {
        return Task.Delay(0);
    }

    public Task OnCompletedAsync()
    {
        return Task.Delay(0);
    }

    public Task OnErrorAsync(Exception ex)
    {
        return Task.Delay(0);
    }
}
```

I am using this against local ASF running single node and azure storage emulator 4.4.

**Comment by charleszipp** [comment]
I was able to get this working by setting the PubSubType to ImplicitOnly and using V2 of AzureQueueStreamProvider on both silo and client. I am not sure why that matters but, i got the idea from #1272.

**Comment by jason-bragg** [comment]
The "Trying to register non active IStreamProducerExtension" exception is unexpected.  That error indicates that the pulling agent responsible for the queue is running on a silo that is not 'active'.

The reason this went away when using ImplicitOnly, is that implicit only pub sub uses local state so there is no need to validate the producers.

I'm glad you were able to resolve this, but let's keep this issue open until we can figure out why the unexpected error occurred.  I suspect this is some sort of timing or ordering issue in our startup logic as it seems like the agents are registering as stream producers even before their silo is are recognized as 'active'.

**Comment by charleszipp** [comment]
thanks @jason-bragg. I went ahead and uploaded the complete solution that repro's the issue. This is using ServiceFabric SDK 5.5. I appreciate you guys being so responsive!
[Audit.OrleansTest.zip](https://github.com/dotnet/orleans/files/1032203/Audit.OrleansTest.zip)

**Comment by rikbosch** [comment]
We are getting the same exceptions, using the azurequeuev2 implementation and servivefabric.

**Comment by sergeybykov** [comment]
Resolved via #3061.

**Comment by ReubenBond** [comment]
#3061 was in response to this, but this is the issue which Jason is working on at the moment. As we discussed this morning, we should aim for a fix in 1.5.0.

**Comment by ReubenBond** [comment]
I should have been more clear in my PR, @sergeybykov, my bad.

**Comment by ReubenBond** [comment]
Resolved by #3128, I tested the fix locally using a private build. That required upgrading the sample project to my 1.5.0 build and changing the client to match the upgraded Service Fabric support:
```C#
namespace Audit.OrleansTest.Terminal
{
    internal class Program
    {
        private static void Main(string[] args)
        {
            Console.WriteLine("Press any key once service is ready...");
            Console.ReadKey();

            var serviceName = new Uri("fabric:/Audit.OrleansTest/Audit.OrleansTest.Service");
            var config = new ClientConfiguration
            {
                DeploymentId = Regex.Replace(serviceName.PathAndQuery.Trim('/'), "[^a-zA-Z0-9_]", "_")
            };
            config.AddAzureQueueStreamProvider("AZQProvider", "UseDevelopmentStorage=true", deploymentId: "orleanstestqueue");
            var builder = new ClientBuilder().UseConfiguration(config).AddServiceFabric(serviceName);
            var client = builder.Build();
            client.Connect().Wait();
            var id = Guid.NewGuid();
            var streamProvider = client.GetStreamProvider("AZQProvider");            
            var stream = streamProvider.GetStream<string>(id, "TESTDATA");
            stream.OnNextAsync("testing123").Wait();
            Console.WriteLine($"Published to stream TESTDATA id {id}");
            Console.ReadKey();
        }
    }
}
```
